### PR TITLE
simplex noise map / 3 Energy to cross ocean tiles

### PR DESCRIPTION
**Pull request by caillef**

### CODE REVIEW

File: client/src/phaser/systems/simplex.js
Comment (ponderingdemocritus) [inline client/src/phaser/systems/simplex.js:1]: do we still need this?

File: contracts/Scarb.toml [hunk 1]
```diff
@@ -8,6 +8,7 @@ sierra-replace-ids = true
 
 [dependencies]
 dojo = { git = "https://github.com/dojoengine/dojo", tag = "v0.3.15" }
+cubit = { git = "https://github.com/influenceth/cubit" }
```
Comment (ponderingdemocritus) [inline contracts/Scarb.toml:11]: can you fix to a version here - good practice to do this

File: contracts/src/actions.cairo [hunk 2]
```diff
@@ -119,13 +119,24 @@ mod actions {
 
             // resolve encounter
             let adversary = player_at_position(world, x, y);
+
+            let tile = tile_at_position(x - ORIGIN_OFFSET.into(), y - ORIGIN_OFFSET.into());
+            let mut moveEnergyCost = MOVE_ENERGY_COST;
```
Comment (ponderingdemocritus) [inline contracts/src/actions.cairo:124]:
```suggestion
            let mut move_energy_cost = MOVE_ENERGY_COST;
```

update to snake everywhere

File: contracts/src/actions.cairo [hunk 3]
```diff
@@ -119,13 +119,24 @@ mod actions {
...
+            if tile == 3 {
+                // Use more energy to go through ocean tiles
+                moveEnergyCost = MOVE_ENERGY_COST * 3;
```
Comment (ponderingdemocritus) [inline contracts/src/actions.cairo:127]:
```suggestion
                move_energy_cost = move_energy_cost * 3;
```

File: client/src/phaser/systems/simplex.js
Comment (caillef) [reply client/src/phaser/systems/simplex.js:1]: I removed this and fixed all the other comment. Thank you for the review, let me know if the changes are good for you

**Comment by ponderingdemocritus** [comment]
amazing work - great improvement. We should keep pushing on this and maybe make unmoveable tiles that are mountains.

**Pull request merged**